### ISSUE TITLE
Implement game generation timeout

### ISSUE THREAD

**Issue by SteffenBauer** [SteffenBauer/PocketPuzzles#18]
As reported by TZornG in #17 , Puzzles with custom game parameters might not be possible for certain parameter values; causing the app to hang in an infinite loop. The app must then be terminated manually via the task manager, which is quite bad.

To be checked how to introduce a timeout; either directly in the generation loop of every game; or high-level by moving game generation to a separate thread. Quick-fix is to introduce more parameter limitation checks in affected puzzles.

**Comment by Skeeve** [comment]
Is there really a loop? Can't it just check on each iteration the time passed and bail out if it was too long?

**Comment by SteffenBauer** [comment]
No, because I get the calculation time for puzzle generation only *after* the game generator finished. Determining of the generation time was too long makes no sense anymore then.
I'm thinking of displaying a button during game generation where pressing it will cancel the game generator. This is more tricky than it sounds because the game generator is blocking.
